Source/JavaScriptCore/builtins/JSIteratorPrototype.js

```diff
@@ -32,10 +32,13 @@ function map(mapper)
     if (!@isObject(this))
         @throwTypeError("Iterator.prototype.map requires that |this| be an Object.");
 
-    if (!@isCallable(mapper))
+    var iterated = this;
```

Comment:
Let's just use `this` directly in this function instead of `iterated` when possible.